src/main/job-queues/job-queue-base.ts

```diff
@@ -77,8 +81,8 @@ export class JobQueueBase implements JobQueue {
                 clearTimeout(this.timeout)
             }
             // eslint-disable-next-line @typescript-eslint/await-thenable
-            const hadSomething = await this.readState()
-            this.timeout = setTimeout(() => this.syncState(), hadSomething ? 0 : this.intervalSeconds * 1000)
+            await this.readState()
```

Comment:
Given existing semantics, `readState` is supposed to finish everything required with the jobs seen so far. So, it doesn't make sense to poll again right after, if it found some jobs to process.

Reply:
The idea here is that if you have 10k jobs to run, you shouldn't wait 5 sec between them (or between every 100 of them) :). So if we got a job with the last run, immediately see if there's another one. Otherwise wait a few seconds and then try again.

Reply:
Ah, that makes sense... assuming there's batching involved. I haven't seen it so far (am I missing something?) - so didn't assume such a state was possible.

Will revert.